File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int n = nums1.length, m = nums2.length;
        int len = n + m;
        int[] merged = new int[len];

        int i = 0, j = 0, k = 0;

        while(i < n && j < m){
            if(nums1[i] <= nums2[j]){
                merged[k] = nums1[i++];
            }else{
                merged[k] = nums2[j++];
            }
            k++;
        }
        while(i < n){
            merged[k++] = nums1[i++];
        }
        while(j < m){
            merged[k++] = nums2[j++];
        }
        int mVal = len / 2; 
        int mid1 = merged[mVal];
        if(len % 2 == 0){
            int mid2 = merged[mVal - 1];
            return (double)(mid1 + mid2) / 2;
        } 
        return (double)mid1;
    }
}

File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.c
double findMedianSortedArrays(int* nums1, int nums1Size, int* nums2, int nums2Size){
    int temp[nums1Size+nums2Size], x = 0;
    for(int i = 0; i < nums1Size + nums2Size; i++){
        if(i < nums1Size){
            temp[i] = nums1[i];
        }else{
            temp[i] = nums2[x++];
        }
    }
    for(int i = 0; i < nums1Size + nums2Size; i++){
        for(int j = i + 1; j < nums1Size + nums2Size; j++){
            if(temp[i] > temp[j]){
                int t = temp[i];
                temp[i] = temp[j];
                temp[j] = t;
            }
        }
    }
    if((nums1Size + nums2Size) % 2 == 0){
        int l = (nums1Size + nums2Size) / 2;
        return (temp[l] + temp[l - 1]) / 2.00;
    }else{
        return temp[(nums1Size + nums2Size) / 2];
    }
}